code/modules/mob/living/brain/brain_item.dm

```diff
@@ -181,32 +181,36 @@
 
 	add_fingerprint(user)
 
-	if(user.zone_selected != BODY_ZONE_HEAD)
+	if(user.zone_selected != zone)
 		return ..()
 
 	var/target_has_brain = C.getorgan(/obj/item/organ/brain)
 
-	if(!target_has_brain && C.is_eyes_covered())
+	if(!target_has_brain && C.is_eyes_covered() && user.zone_selected == BODY_ZONE_HEAD)
 		to_chat(user, span_warning("You're going to need to remove [C.p_their()] head cover first!"))
 		return
 
+	// This should be a better check but this covers 99.9% of cases
+	if((C?.dna?.species?.inherent_biotypes & MOB_ROBOTIC && status != ORGAN_ROBOTIC) || (C?.dna?.species?.inherent_biotypes & MOB_ORGANIC && status != ORGAN_ORGANIC))
+		to_chat(user, span_warner("This brain is incompatiable with this beings biology!"))
+		return
+
 //since these people will be dead M != usr
 
 	if(!target_has_brain)
-		if(!C.get_bodypart(BODY_ZONE_HEAD) || !user.temporarilyRemoveItemFromInventory(src))
+		if(!C.get_bodypart(zone) || !user.temporarilyRemoveItemFromInventory(src))
 			return
-		var/msg = "[C] has [src] inserted into [C.p_their()] head by [user]."
+		var/msg = "[C] has [src] inserted into [C.p_their()] by [user]."
 		if(C == user)
-			msg = "[user] inserts [src] into [user.p_their()] head!"
+			msg = "[user] inserts [src] into [user.p_their()]!"
```

Comment:
into them

Reply:
Whoop will fix tomorrow unless you give me a change request block

Reply:
> Whoop will fix tomorrow unless you give me a change request block

i can't i am quite powerlessa ctually

Reply:
maybe i could try..

Reply:
Didnt know i could edit on mobile was painful though tabs dont exist